src/registry.rs

```diff
@@ -41,7 +41,21 @@ pub struct Registry {
 	string_table: Interner<&'static str>,
 	#[serde(skip)]
 	type_table: Interner<AnyTypeId>,
-	types: Vec<TypeIdDef>,
+    #[serde(serialize_with = "serialize_registry_types")]
+	types: BTreeMap<UntrackedSymbol<std::any::TypeId>, TypeIdDef>,
+}
+
+/// Serializes the types of the registry by removing their unique IDs
+/// and instead serialize them in order of their removed unique ID.
```

Comment:
It is hard for me to understand this comment.
Something like this?
Serializes the types in the registry without their unique IDs and the serialization order is based on the unique ID.
"Unique ID" sounds very vague to me. Just call it "rust type ID"?

Reply:
It isn't ordered by the Rust type ID but by the untracked symbol ID generated by the registry upon registering a type initially. The ordering before this PR was kind off reversed and I could have simply reversed to order of the vector but this change actually makes it much more clear what needs to be done.
I will improve the comment.